internal/service/sagemaker/model_test.go

```diff
@@ -955,3 +1068,237 @@ resource "aws_security_group" "test" {
+    eu-west-2      = "763104351884"
+    eu-west-3      = "763104351884"
+    eu-north-1     = "763104351884"
+    eu-south-1     = "692866216735"
+    eu-south-2     = "503227376785"
+    il-central-1   = "780543022126"
+    me-south-1     = "217643126080"
+    me-central-1   = "914824155844"
+    sa-east-1      = "763104351884"
+    ap-southeast-1 = "763104351884"
+    ap-southeast-2 = "763104351884"
+    ap-southeast-3 = "907027046896"
+    ap-southeast-4 = "457447274322"
+    ap-northeast-1 = "763104351884"
+    ap-northeast-2 = "763104351884"
+    ap-northeast-3 = "364406365360"
+    ap-south-1     = "763104351884"
+    ap-south-2     = "772153158452"
+    ap-east-1      = "871362719292"
+    sa-east-1      = "763104351884"
+    cn-north-1     = "727897471807"
+    cn-northwest-1 = "727897471807"
+  }
+  account = local.region_account_map[data.aws_region.current.name]
+  primary_container_image = format(
+    "%%s.dkr.ecr.%%s.%%s/huggingface-pytorch-tgi-inference:2.0.1-tgi1.1.0-gpu-py39-cu118-ubuntu20.04",
+    local.account,
+    data.aws_region.current.name,
+    data.aws_partition.current.dns_suffix
+    
+  primary_container_model_data_source_s3_uri = format(
+    "s3://jumpstart-private-cache-prod-%%s/meta-textgeneration/meta-textgeneration-llama-2-13b-f/artifacts/inference-prepack/v1.0.0/",
+    data.aws_region.current.name
+  )
+}
+
+resource "aws_sagemaker_model" "test" {
+  name                     = %[1]q
+  enable_network_isolation = true
+  execution_role_arn       = aws_iam_role.test.arn
+  
+  primary_container {
+    image = local.primary_container_image
+    mode = "SingleModel"
+    environment = {
+      ENDPOINT_SERVER_TIMEOUT = 3600
+      HF_MODEL_ID = "/opt/ml/model"
+      MAX_INPUT_LENGTH = 4095
+      MAX_TOTAL_TOKENS = 4096
+      MODEL_CACHE_ROOT = "/opt/ml/model"
+      SAGEMAKER_ENV = 1
+      SAGEMAKER_MODEL_SERVER_WORKERS = 1
+      SAGEMAKER_PROGRAM = "inference.py"
+      SM_NUM_GPUS = 4
+  }
+
+  model_data_source {
+    s3_data_source {
+      compression_type = "None"
+      s3_data_type     = "S3Prefix"
+      s3_uri           = local.primary_container_model_data_source_s3_uri
+      model_access_config {
+        accept_eula = true
+      }
+    }
+  }
+}
+`, rName))
+}
+
+func testAccModelConfig_primaryContainerInferenceSpecificationName(rName string) string {
+	return acctest.ConfigCompose(testAccModelConfig_base(rName), fmt.Sprintf(`
+
+resource "aws_sagemaker_model" "test" {
+  name               = %[1]q
+  execution_role_arn = aws_iam_role.test.arn
+
+  primary_container {
+    image                        = data.aws_sagemaker_prebuilt_ecr_image.test.registry_path
+    inference_specification_name = "test"
+  }
+}
+`, rName))
+}
+
+func testAccModelConfig_primaryContainerMultiModelConfigModelCacheSetting(rName string) string {
+	return acctest.ConfigCompose(testAccModelConfig_base(rName), fmt.Sprintf(`
+data "aws_region" "current" {}
+data "aws_partition" "current" {}
+locals {
```

Comment:
consider leveraging prebuilt_ecr_image datasource

Reply:
Requires a change in data source to include these new images ex. huggingface-pytorch-tgi , Hence had to fallback on url parsing with accounts